src/simplify_algebra.cpp

```diff
@@ -445,6 +477,38 @@ struct find_mul_add
         m.replace_instruction(ins, make_op("add"), ax_ins, ab_ins);
     }
 };
+*/
+
+struct find_scalar_mul_conv
+{
+    auto matcher() const
+    {
+        return match::name("mul")(
+            match::either_arg(0, 1)(
+                match::is_constant().bind("scalar"),
+                match::name("convolution").bind("conv")
+            ));
```

Comment:
This should be checking for slice -> mul -> add. There is no reason to check for convolution.

Reply:
Also, in the `apply` method you need to check the other slices go into add or mul -> add.